website/blog/2023-02-10.ferretdb-compare-arrays.md

```diff
@@ -0,0 +1,605 @@
+  { _id: 'number', v: 4, op: true }
+])
+```
+
+In this example `{$max: {v: ['c']}}` is used
+and this updates document if specified value `['c']` is greater than the existing
+value in the field `v` of documents.
+
+
+The records updated were `array-string` and `number`. `array-string` was updated by
+iterating the first index of the array, and `'c'` was greater than existing value of `'b'`.
+was used to compare array type of the update `number` was updated because array `['c']` is greater
+in BSON comparison order than `number`.
+
+```js
+db.updatearray.updateMany({op:true},{$max: {v: ['c']}})
+{
+  acknowledged: true,
+  insertedId: null,
+  matchedCount: 4,
+  modifiedCount: 2,
+  upsertedCount: 0
+}
+db.updatearray.find()
+[
+  { _id: 'bool', v: true, op: true },
+  { _id: 'array-string', v: [ 'c' ], op: true },
+  { _id: 'array-mixed', v: [ 'g', 1 ], op: true },
+  { _id: 'number', v: [ 'c' ], op: true }
+]
+```
+
+* `bool` was not updated because array `['c']` is not greater than boolean 
+  `true` in BSON comparison order.
+* `array-string` was updated because `'c'` was greater than the first index of array `'b'`.
+* `array-mixed` was not updated because `'c'` was not greater than `'g'`.
+* `number` was updated because number is not greater than array in BSON comparison order.
+
+### Update Operator with Nested Array
+
+Let's add nested array to use `$max` operator to update the collection.
+```js
+db.updatenested.insertMany([
+  { _id: 'bool', v: [ [ true ] ], op: true },
+  { _id: 'array-string', v: [ [ 'b', 'g' ] ], op: true },
+  { _id: 'array', v: [ [ 'g', 1 ] ], op: true },
+  { _id: 'number', v: [ 4 ], op: true }
+])
+```
+
+Here we use array update operator `{$max: {v: ['c']}}`, because comparison of array and
+other type is where comparison behaves differently.
+
+```js
+db.updatenested.updateMany({op:true},{$max: {v: ['c']}})
+{
+  acknowledged: true,
+  insertedId: null,
+  matchedCount: 4,
+  modifiedCount: 1,
+  upsertedCount: 0
+}
+db.updatenested.find()
+[
+  { _id: 'bool', v: [ [ true ] ], op: true },
+  { _id: 'array-string', v: [ [ 'b', 'g' ] ], op: true },
+  { _id: 'array', v: [ [ 'g', 1 ] ], op: true },
+  { _id: 'number', v: [ 'c' ], op: true }
+]
+```
+
+`number` was updated because first index of `['c']`, string `'c'` has greater comparison order 
+than number `4`.
+
+No nested array were updated since array has greater BSON comparison order than string.
+
+From this example, we understand that `$max` purely uses BSON comparison order for
+comparing other BSON types and array does not take special role.
+
+## Why Nested Array is not Supported at FerretDB
+
+At FerretDB, we do not support nested array at the moment.
+We did not initially support nested array due to complexities and
+difference in comparison of
+[array with non-array BSON type](#compare-array-and-other-bson-type).
+There is uncertainty on how filter such as `$gt` works on [nested array](#nested-array-with-gt)
+comparison.
```

Comment:
This entire section can probably be phrased better, summarizing the entire article and how we can don't support nested arrays at the moment.

Reply:
I've expanded this more. Feel free to suggest :hugs: